src/components/SendTransactionFlow.tsx

```diff
@@ -0,0 +1,350 @@
+                  ...formData,
+                  memoType: e.target.value,
+                });
+              }}
+              value={formData.memoType}
+            >
+              <option value={StellarSdk.MemoText}>MEMO_TEXT</option>
+              <option value={StellarSdk.MemoID}>MEMO_ID</option>
+              <option value={StellarSdk.MemoHash}>MEMO_HASH</option>
+              <option value={StellarSdk.MemoReturn}>MEMO_RETURN</option>
+            </TempSelectInputEl>
+          </El>
+          <El>
+            Memo Content:{" "}
+            <TempInputEl
+              type="text"
+              placeholder={memoPlaceholderMap[formData.memoType]}
+              onChange={(e) => {
+                onInput({
+                  ...formData,
+                  memoContent: e.target.value,
+                });
+              }}
+              value={formData.memoContent}
+            ></TempInputEl>
+          </El>
+          <El>
+            <TempAnchorEl
+              onClick={() => {
+                onInput({
+                  ...formData,
+                  memoType: StellarSdk.MemoNone,
+                  memoContent: "",
+                });
+                setIsMemoVisible(false);
+              }}
+            >
+              Remove memo:
+            </TempAnchorEl>
+          </El>
+        </>
+      )}
+      <El>
+        Fee (lumens) :{" "}
+        <TempInputEl
+          type="number"
+          value={formData.fee}
+          onChange={(e) => {
+            onInput({ ...formData, fee: e.target.value });
+          }}
+        ></TempInputEl>
+      </El>
+      <button onClick={props.onContinue}>Continue</button>
+    </El>
+  );
+};
+
+interface ConfirmProps {
+  onSuccessfulTx: () => void;
+  onFailedTx: () => void;
+  formData: FormData;
+}
+
+const ConfirmTransaction = (props: ConfirmProps) => {
+  const { sendTx, keyStore } = useRedux(["sendTx", "keyStore"]);
+  const { formData, onSuccessfulTx, onFailedTx } = props;
+  const dispatch = useDispatch();
+
+  const createMemo = (memoType: string, memoContent: string) => {
+    switch (memoType) {
+      case StellarSdk.MemoText:
+        return StellarSdk.Memo.text(memoContent);
+      case StellarSdk.MemoID:
+        return StellarSdk.Memo.id(memoContent);
+      case StellarSdk.MemoHash:
+        return StellarSdk.Memo.hash(memoContent);
+      case StellarSdk.MemoReturn:
+        return StellarSdk.Memo.return(memoContent);
+      case StellarSdk.MemoNone:
+      default:
+        return StellarSdk.Memo.none();
+    }
+  };
+
+  const handleSend = async () => {
+    const { privateKey } = await loadPrivateKey(keyStore.id, keyStore.password);
+    const result = await dispatch(
+      sendTxAction({
+        secret: privateKey,
+        toAccountId: formData.toAccountId,
+        amount: formData.amount,
+        // Round to nearest Stroom
+        fee: Math.round(Number(formData.fee) * 1e7),
+        memo: createMemo(formData.memoType, formData.memoContent),
```

Comment:
To make it a little easier, we could move the `memo` creation to the `TransactionBuilder`. That way we could keep `memo` and `memoType` separate until the very last step. What do you think?